osu.Game/Overlays/Mods/ModSelectOverlay.cs

```diff
@@ -371,8 +371,11 @@ protected override void LoadComplete()
         {
             base.LoadComplete();
 
-            SelectedMods.BindValueChanged(_ => updateSelectedButtons());
             availableMods.BindValueChanged(_ => updateAvailableMods(), true);
+
+            // intentionally bound after the above line to avoid a potential update feedback cycle.
+            // i haven't actually observed this happening but as updateAvailableMods() changes the selection it is plausible.
+            SelectedMods.BindValueChanged(_ => updateSelectedButtons());
```

Comment:
And it's fine if this doesn't run initially, as it was doing prior to 435c85a2e79b8682d093c261054c8d4ad67215b4?

Reply:
It's done in the line before it (see the method it calls - this is the last thing it does).

Reply:
Yeah I get that. My concern is that `updateSelectedButtons` has the `.ToList()` call there to avoid mutating `SelectedMods` during iteration (which does happen), and that if this doesn't run initially after that there could be a point where mods aren't updated after that mutation happens. As for what issue that would cause - I don't know, it's just a point of consideration.

Reply:
Ah. I don't believe that can happen... unless there was a weird threading issue (out of scope).